pontoon/sync/changeset.py

```diff
@@ -234,24 +238,27 @@ def update_entity_translations_from_vcs(
                 ))
 
         # Any existing translations that were not approved get unapproved.
-        if old_translations is None:
-            old_translations = db_translations.filter(approved_date__lte=self.now)
-
         for translation in old_translations:
             if translation not in approved_translations:
+                translation = (
+                    match_attr(self.translations_to_update, pk=translation.pk) or translation
+                )
                 translation.approved = False
                 translation.approved_user = None
                 translation.approved_date = None
 
-                if translation.is_dirty():
+                if translation not in self.translations_to_update and translation.is_dirty():
```

Comment:
I have a non-blocking comment: If we'd change a type of translations_to_update into something dict/set like (with translation_pk as the index) that could simplify those changes.